### ISSUE TITLE
Closing Popover Dynamically / After Click...

### ISSUE THREAD

**Issue by jeffconklin** [sandywalker/webui-popover#124]
I really like this plugin and use it often. But I really need an easy way to "Close" a popover that is already opened when someone clicks on an element in that popover. A simple example is that a user clicks a button that shows the popover. That popover contains a list of links to other pages or actions like displaying a modal window. When they click an action to display a modal, I could not easily find a way to close the popover and then show the modal.

Though I know I can call destroy the popover element by calling  $(element).webuiPopover('destroy').webuiPopover(settings), that is just not feasible when settings may be different in different situations. I also don't want to ALWAYS destroy and create all the popovers when all I need is for the popover to close.

I searched around for an efficient solution for this, but could not find one. So I put together something that I feel is a hack, but it works... 

https://jsbin.com/cumigi/edit?html,js,output

In short, my flow is:
- Append a hidden div on pageload to the body with a specific ID: close-webui
- Attach webuiPopover to all the required elements, including the hidden div
- When I call ANY function (like displaying a modal) when the popover is already displayed, I call $('#close-webui').webuiPopover('hideAll');

This will hide all the popovers without having to destroy and recreate them.

Question for the author: Is calling 'hideAll' a valid approach? I do not like using undocumented features with my production solutions. But the thought of destroying all my popovers and recreating them JUST to close the current one that I am interacting with seems overkill.

Thanks.

**Comment by sandywalker** [comment]
Sorry for replayed so late, your solution seems OK although not elegant. You can call 'hideAll' without any negative effects, but why not just call 'hide' on trigger element?

**Comment by jeffconklin** [comment]
"You can call 'hideAll' without any negative effects, but why not just call 'hide' on trigger element?"

* Short answer: I do not know how to determine the original element to call 'hide'.
* Longer answer:
Calling $('#close-webui').webuiPopover('hideAll'); has been the only predictable way that I can close any opened popovers for any situation. As displayed in my JSBin example, the original button does not have an ID. Once anything inside the generated popover is clicked, how can I determine which element/popover it is associated to? I was not able to see an "easy" way to determine this. Though I can add an ID to the original button change how I create the webuiPopover initially, the generate content in the popover will also need to know the ID. This changes the scope of what my scripts need to keep track of, and is why I started using $('#close-webui').webuiPopover('hideAll');

I think a global "hide all" function would be a great addition. I would no longer have to generate a fake element (like I have now) to accomplish the same thing.

Also, is there an "easy" way to determine which popovers are currently displayed/active?

Thank you for the reply. And thank you in advance for any help.

**Comment by sandywalker** [comment]
Thanks for your reply, I got what you mean. 
The element in the popover content can't get the popover info vary easy , I think  the best way to accomplish  is by adding global function ,may be there is better way, I don't know. I'll try to add this feature.

**Comment by sandywalker** [comment]
I've added this feature.
